virtualfish/virtual.fish

```diff
@@ -129,13 +169,92 @@ function __vf_new --description "Create a new virtualenv"
     end
 
     emit virtualenv_will_create
-    set envname $argv[-1]
-    set -e argv[-1]
-    if set -q VIRTUALFISH_DEFAULT_PYTHON
-        set argv "--python" $VIRTUALFISH_DEFAULT_PYTHON $argv
+    argparse -n "vf new" -x q,v,d --ignore-unknown "h/help" "q/quiet" "v/verbose" "d/debug" "p/python=" -- $argv
+
+    if set -q _flag_help
+        echo "Purpose: Creates a new virtual environment"
+        echo "Usage: "(set_color green)"vf new "(set_color -di)"[-p <python-version>] [<virtualenv-flags>]"(set_color normal)(set_color green)" <virtualenv-name>"(set_color normal)
```

Comment:
I like saving the output of `set_color` if I use a color a few times.

Like

```fish
set -l normal (set_color normal)
set -l green (set_color green)
echo "Usage: "$green"vf new "(set_color -di)"[-p <python-version>] [<virtualenv-flags>]"$normal$green" <virtualenv-name>"
```

(opening those quotes for $green to avoid looking for $greenvf)